File: SignalKitTests/Fakes/MockControl.m
//
//  MockControl.m
//  SignalKit
//
//  Created by Yanko Dimitrov on 8/12/15.
//  Copyright © 2015 Yanko Dimitrov. All rights reserved.
//

#import <Foundation/Foundation.h>
#import "MockControl.h"

@implementation MockControl

-(void)sendAction:(SEL)action to:(id)target forEvent:(UIEvent *)event {
    
    //
    // Thanks to ReactiveCocoa!
    // https://github.com/ReactiveCocoa/ReactiveCocoa
    //
    #pragma clang diagnostic push
    #pragma clang diagnostic ignored "-Warc-performSelector-leaks"
    
    [target performSelector:action withObject:nil];
    
    #pragma clang diagnostic pop
}

@end

@implementation MockButton

-(void)sendAction:(SEL)action to:(id)target forEvent:(UIEvent *)event {
    
    MockControl* control = [[MockControl alloc] init];
    
    [control sendAction:action to:target forEvent:event];
}

@end

@implementation MockDatePicker

-(void)sendAction:(SEL)action to:(id)target forEvent:(UIEvent *)event {
    
    MockControl* control = [[MockControl alloc] init];
    
    [control sendAction:action to:target forEvent:event];
}

@end

@implementation MockSegmentedControl

-(void)sendAction:(SEL)action to:(id)target forEvent:(UIEvent *)event {
    
    MockControl* control = [[MockControl alloc] init];
    
    [control sendAction:action to:target forEvent:event];
}

@end

@implementation MockSlider

-(void)sendAction:(SEL)action to:(id)target forEvent:(UIEvent *)event {
    
    MockControl* control = [[MockControl alloc] init];
    
    [control sendAction:action to:target forEvent:event];
}

@end

@implementation MockSwitch

-(void)sendAction:(SEL)action to:(id)target forEvent:(UIEvent *)event {
    
    MockControl* control = [[MockControl alloc] init];
    
    [control sendAction:action to:target forEvent:event];
}

@end

@implementation MockTextField

-(void)sendAction:(SEL)action to:(id)target forEvent:(UIEvent *)event {
    
    MockControl* control = [[MockControl alloc] init];
    
    [control sendAction:action to:target forEvent:event];
}

@end

File: SignalKitTests/Fakes/MockControl.swift
//
//  MockControl.swift
//  SignalKit
//
//  Created by Yanko Dimitrov on 3/5/16.
//  Copyright © 2016 Yanko Dimitrov. All rights reserved.
//

import UIKit

class MockControl: UIControl {
    
    override func sendAction(_ action: Selector, to target: Any?, for event: UIEvent?) {
        
        if let target = target as? NSObject {
            
            target.perform(action, with: nil)
        }
    }
}

class MockDatePicker: UIDatePicker {
    
    override func sendAction(_ action: Selector, to target: Any?, for event: UIEvent?) {
        
        let control = MockControl()
        
        control.sendAction(action, to: target, for: event)
    }
}

class MockSegmentedControl: UISegmentedControl {
    
    override func sendAction(_ action: Selector, to target: Any?, for event: UIEvent?) {
        
        let control = MockControl()
        
        control.sendAction(action, to: target, for: event)
    }
}

class MockSlider: UISlider {
    
    override func sendAction(_ action: Selector, to target: Any?, for event: UIEvent?) {
        
        let control = MockControl()
        
        control.sendAction(action, to: target, for: event)
    }
}

class MockSwitch: UISwitch {
    
    override func sendAction(_ action: Selector, to target: Any?, for event: UIEvent?) {
        
        let control = MockControl()
        
        control.sendAction(action, to: target, for: event)
    }
}

class MockTextField: UITextField {
    
    override func sendAction(_ action: Selector, to target: Any?, for event: UIEvent?) {
        
        let control = MockControl()
        
        control.sendAction(action, to: target, for: event)
    }
}
